src/lib/features/unique-connection/unique-connection-service.ts

```diff
@@ -0,0 +1,98 @@
+import type { IUnleashConfig } from '../../types/option';
+import type { IFlagResolver, IUnleashStores } from '../../types';
+import type { Logger } from '../../logger';
+import type { IUniqueConnectionStore } from './unique-connection-store-type';
+import HyperLogLog from 'hyperloglog-lite';
+import type EventEmitter from 'events';
+import { SDK_CONNECTION_ID_RECEIVED } from '../../metric-events';
+
+const n = 12;
+
+export class UniqueConnectionService {
+    private logger: Logger;
+
+    private uniqueConnectionStore: IUniqueConnectionStore;
+
+    private flagResolver: IFlagResolver;
+
+    private eventBus: EventEmitter;
+
+    private activeHour: number;
+
+    private hll = HyperLogLog(n);
+
+    constructor(
+        {
+            uniqueConnectionStore,
+        }: Pick<IUnleashStores, 'uniqueConnectionStore'>,
+        config: Pick<IUnleashConfig, 'getLogger' | 'flagResolver' | 'eventBus'>,
+    ) {
+        this.uniqueConnectionStore = uniqueConnectionStore;
+        this.logger = config.getLogger('services/unique-connection-service.ts');
+        this.flagResolver = config.flagResolver;
+        this.eventBus = config.eventBus;
+        this.activeHour = new Date().getHours();
+    }
+
+    listen() {
+        this.eventBus.on(SDK_CONNECTION_ID_RECEIVED, this.count.bind(this));
+    }
+
+    count(connectionId: string) {
+        if (!this.flagResolver.isEnabled('uniqueSdkTracking')) return;
+        this.hll.add(HyperLogLog.hash(connectionId));
+    }
+
+    async getStats() {
+        const [previous, current] = await Promise.all([
+            this.uniqueConnectionStore.get('previous'),
+            this.uniqueConnectionStore.get('current'),
+        ]);
+        const previousHll = HyperLogLog(n);
+        if (previous) {
+            previousHll.merge({ n, buckets: previous.hll });
+        }
+        const currentHll = HyperLogLog(n);
+        if (current) {
+            currentHll.merge({ n, buckets: current.hll });
+        }
+        return { previous: previousHll.count(), current: currentHll.count() };
+    }
+
+    async sync(clock = () => new Date()): Promise<void> {
```

Comment:
Why use a function instead of a value here? Not that I mind; I'm just curious whether there's a specific reason?